tests/jballerina-unit-test/src/test/resources/test-src/query/query-negative.bal

```diff
@@ -384,4 +384,19 @@ public function testMethodParamInQuery(int age) {
                    lastName: lastName,
                    age: age
              };
-}
\ No newline at end of file
+}
+
+type TableRecord record {
+    readonly string name;
+    int id;
+};
+
+function testMultipleWhereClausesWithSimpleVariable() {
+
+    table<TableRecord> key(name) t = table [
+            {name: "Amy", id: 1234},
+            {name: "John", id: 4567}
+        ];
+
+    table<int> ids = from var x in t select x.id;
```

Comment:
@gimantha can you add a test case with record binding pattern (similar to the example in the issue) ?

Reply:
@mohanvive updated